validation/pipeline/qase/qase.go

```diff
@@ -0,0 +1,174 @@
+func GetTestSuite(project, suite string) (qase.Suite, error) {
+	logrus.Debugf("Getting test suite \"%s\" in project %s\n", suite, project)
+	localVarOptionals := &qase.SuitesApiGetSuitesOpts{
+		FiltersSearch: optional.NewString(suite),
+	}
+	qaseSuites, _, err := Client.SuitesApi.GetSuites(context.TODO(), project, localVarOptionals)
+	if err != nil {
+		return qase.Suite{}, err
+	}
+
+	for _, qaseSuite := range qaseSuites.Result.Entities {
+		if qaseSuite.Title == suite {
+			return qaseSuite, nil
+		}
+	}
+	return qase.Suite{}, fmt.Errorf("test suite \"%s\" not found in project %s", suite, project)
+}
+
+// CreateTestSuite retrieves a Test Suite by name within a specified Qase Project if it exists
+func CreateTestSuite(project, suite string) (int64, error) {
+	logrus.Debugf("Creating test suite \"%s\" in project %s\n", suite, project)
+	suiteBody := qase.SuiteCreate{Title: suite}
+	resp, _, err := Client.SuitesApi.CreateSuite(context.TODO(), suiteBody, project)
+	if err != nil {
+		return 0, fmt.Errorf("failed to create test suite: \"%s\". Error: %v", suite, err)
+	}
+	return resp.Result.Id, nil
+}
+
+// UploadTests either creates new Test Cases and their associated Suite or updates them if they already exist
+func UploadTests(project string, testCases []TestCase) error {
+	for _, tc := range testCases {
+		logrus.Info("Uploading test case:\n\tProject: ", project, "\n\tTitle: ", tc.Title, "\n\tDescription: ", tc.Description, "\n\tSuiteId: ", tc.SuiteId, "\n\tAutomation: ", tc.Automation, "\n\tSteps: ", tc.Steps)
+
+		existingCase, err := GetTestCase(project, tc.Title)
+		if err == nil {
+			var qaseTest qase.TestCaseUpdate
+			qaseTest.Title = tc.Title
+			qaseTest.Description = tc.Description
+			qaseTest.SuiteId = tc.SuiteId
+			qaseTest.Automation = tc.Automation
+			for _, step := range tc.Steps {
+				var qaseSteps qase.TestCaseUpdateSteps
+				qaseSteps.Action = step.Action
+				qaseSteps.ExpectedResult = step.ExpectedResult
+				qaseSteps.Data = step.InputData
+				qaseSteps.Position = step.Position
+				qaseTest.Steps = append(qaseTest.Steps, qaseSteps)
+			}
+			err = updateTestCases(project, qaseTest, int32(existingCase.Id))
+			if err != nil {
+				return err
+			}
+		} else {
+			var qaseTest qase.TestCaseCreate
+			qaseTest.Title = tc.Title
+			qaseTest.Description = tc.Description
+			qaseTest.SuiteId = tc.SuiteId
+			qaseTest.Automation = tc.Automation
+			for _, step := range tc.Steps {
+				var qaseSteps qase.TestCaseCreateSteps
+				qaseSteps.Action = step.Action
+				qaseSteps.ExpectedResult = step.ExpectedResult
+				qaseSteps.Data = step.InputData
+				qaseSteps.Position = step.Position
+				qaseTest.Steps = append(qaseTest.Steps, qaseSteps)
+			}
+			err = createTestCases(project, qaseTest)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func createTestCases(project string, testCase qase.TestCaseCreate) error {
```

Comment:
since this only creates one test case, rename to `createTestCase`

Reply:
also, do these not get added to any particular test suite?

Reply:
They do! That's part of the TestCaseCreate data -- a `SuiteId` field.